lib/chef/provider/service/macosx.rb

```diff
@@ -41,11 +44,27 @@ def self.gather_plist_dirs
         PLIST_DIRS = gather_plist_dirs
 
         def load_current_resource
-          @current_resource = Chef::Resource::Service.new(@new_resource.name)
+          @current_resource = Chef::Resource::MacosxService.new(@new_resource.name)
           @current_resource.service_name(@new_resource.service_name)
           @plist_size = 0
-          @plist = find_service_plist
+          @plist = @new_resource.plist ? @new_resource.plist : find_service_plist
+          Chef::Log.debug("Plist: '#{@plist}'")
           @service_label = find_service_label
+          Chef::Log.debug("Service_Label: '#{@service_label}'")
```

Comment:
It's nice in providers if these debug messages include resource so you can more easily identify the log message you're looking for. This is an example of the common pattern:

```
Chef::Log.debug("#{new_resource} service_label: '#{@service_label}'")
```